### ISSUE TITLE
Add contextual help tooltips with docs links throughout the UI

### ISSUE THREAD

**Issue by simeonlee** [tensorzero/tensorzero#6279]
## Summary

Add contextual help (?) icons throughout the UI that show brief explanations on hover and optionally link to relevant documentation pages. This improves discoverability and reduces the need to leave the app to look up concepts.

## Approach

A shared `HelpTooltip` component standardizes the pattern:

```tsx
<HelpTooltip docsPath="gateway/guides/episodes">
  Episodes group related inferences from multi-step workflows.
</HelpTooltip>
```

- Light gray `CircleHelp` icon (`text-muted-foreground`, 14px)
- Tooltip appears on hover with brief explanation
- When `docsPath` is provided, clicking the icon opens the docs page in a new tab
- `SectionHeader` now accepts a `help` prop for easy integration with section headings

### Design principles

- **Selective, not exhaustive** — only where concepts are genuinely non-obvious
- **Brief and actionable** — 1-2 sentences, not documentation dumps
- **Co-located content** — tooltip text lives next to the UI element it explains (no centralized constants file that creates indirection)
- **Consistent pattern** — same component everywhere, same icon size, same interaction

### High priority (most confusing to new users)

| Page | Element | Tooltip content | Docs link |
|------|---------|----------------|-----------|
| **SFT form** | Validation Split (%) | Percentage of training data reserved for validation to detect overfitting | `optimization/supervised-fine-tuning-sft` |
| **SFT form** | Max. Samples | Maximum training examples to use, limited by curated inferences meeting threshold | `optimization/supervised-fine-tuning-sft` |
| **SFT form** | Metric/threshold (curation) | Minimum metric score for an inference to be included as training data | `optimization/supervised-fine-tuning-sft` |
| **Evaluation modal** | Inference Cache | Controls whether evaluation results are cached/reused | `gateway/guides/inference-caching` |
| **Evaluation modal** | Concurrency | Number of datapoints evaluated in parallel (already existed, now uses shared component) | — |
| **Evaluation modal** | Adaptive Stopping Precision | Stop evaluator when confidence interval is within threshold (already existed) | — |
| **Function detail** | Experimentation section | How traffic is distributed across variants via weights | `experimentation/run-static-ab-tests` |
| **Episodes page** | Page header | What episodes are (multi-step workflow sequences) | `gateway/guides/episodes` |
| **Models page** | Latency section | Explanation of P50/P90/P95 percentile metrics | — |
| **Dataset builder** | Output source selector | Difference between None/Inference/Demonstration outputs | `gateway/api-reference/datasets-datapoints` |

### Medium priority

| Page | Element | Tooltip content | Docs link |
|------|---------|----------------|-----------|
| **Function detail** | Variant weights in table | Percentage of traffic sent to this variant | `experimentation/run-static-ab-tests` |
| **Function detail** | Throughput section | Number of inference requests per time period | — |
| **Workflow evaluations** | Page header | How workflow evaluations differ from inference evaluations | `evaluations/workflow-evaluations/tutorial` |
| **Inference detail** | Model Inferences section | Why one inference can produce multiple model inferences | — |
| **Dataset builder** | Episode ID filter | Filter to inferences from a specific multi-step sequence | `gateway/guides/episodes` |
| **Playground** | Reference Output | Expected output from the dataset for comparison | — |
| **API keys** | Public ID column | The non-secret identifier for API key usage | `operations/set-up-auth-for-tensorzero` |

### Lower priority (straightforward but nice to have)

| Page | Element | Tooltip content | Docs link |
|------|---------|----------------|-----------|
| **Models page** | Usage section | Number of inference requests to each model over time | — |
| **Models/Functions** | Time granularity selector | How metrics are aggregated over time | — |
| **Inference detail** | Demonstration feedback | Human-provided reference outputs used to guide model behavior | `gateway/guides/metrics-feedback` |
| **Function detail** | Schemas section | Input/output schemas defining the function API contract | `gateway/configure-functions-and-variants` |
| **Datasets** | Page header | What datasets are and how they're used for evaluation/optimization | `gateway/api-reference/datasets-datapoints` |

## What NOT to add help tooltips to

- **Self-explanatory labels** — "Function", "Variant", "Time", "ID" don't need explanation
- **Standard UI patterns** — search bars, pagination, sort controls
- **Every column header** — only where the metric/concept is ambiguous
- **Sidebar navigation** — page names are clear enough

## Implementation notes

- The `HelpTooltip` component is already built and the initial batch is implemented in the companion PR
- New tooltips can be added incrementally per-page without coordination
- No centralized constants file — each tooltip text is co-located with its usage, keeping code greppable and avoiding a layer of indirection
- Docs base URL (`https://www.tensorzero.com/docs`) is defined once in the component
